src/ObjectPool/src/DefaultObjectPool.cs

```diff
@@ -10,10 +10,10 @@ namespace Microsoft.Extensions.ObjectPool
 {
     public class DefaultObjectPool<T> : ObjectPool<T> where T : class
     {
-        private readonly ObjectWrapper[] _items;
+        protected readonly ObjectWrapper[] _items;
```

Comment:
hmmm, I'd really like to avoid this showing up in a public API surface. Move to a top level struct and mark internal?

Reply:
That would accomplish roughly the same thing so that's fine. I just want to make sure that implementation details like don't leak out. It's probably appropriate to make `DisposableObjectPool<T>` internal as well.

Reply:
Done -- will be updated with the next push.